async_service/function_service/async_function_deployment.py

```diff
@@ -0,0 +1,154 @@
+                )
+            )
+        )
+
+        # Build and configure the applications
+        server_app = async_func_deployment.build_async_server_app()
+        worker_app = async_func_deployment.build_worker_app()
+
+        # These two apps can now be run on different ports or different machines.
+    """
+
+    def __init__(
+        self,
+        worker_config: WorkerConfig,
+        functions: Dict[str, Callable],
+        init_function: Callable = None,
+    ) -> None:
+        if (
+            INTERNAL_FUNCTION_NAME in functions
+            or FUNCTION_SCHEMA_ENDPOINT.lstrip("/") in functions
+            or RESULT_ENDPOINT.split("/")[1] in functions
+        ):
+            raise ValueError(
+                f"Function names {INTERNAL_FUNCTION_NAME},  {FUNCTION_SCHEMA_ENDPOINT.lstrip('/')} and RESULT_ENDPOINT.split('/')[1] are reserved for internal use."
+            )
+        self.functions = functions
+        self.worker_config = worker_config
+        self.init_function = init_function
+
+    def register_function(self, name: str, function: Callable) -> None:
+        self.functions[name] = function
+
+    def build_worker_app(self) -> FastAPI:
+        app = FastAPI(root_path=os.getenv("TFY_SERVICE_ROOT_PATH"), docs_url="/")
+
+        functions = self.functions
+        init_function = self.init_function
+
+        class FunctionProcessor(Processor):
+            def init(self):
+                if init_function:
+                    init_function()
+
+            def process(self, input_message: InputMessage) -> int:
+                body = input_message.body
+                func_name = body.pop(INTERNAL_FUNCTION_NAME, None)
+                if func_name is None:
+                    raise ValueError(
+                        f"Input message does not contain {INTERNAL_FUNCTION_NAME} key."
+                    )
+
+                func = functions[func_name]
+                return func(**body)
+
+        app = FunctionProcessor().build_app(worker_config=self.worker_config)
+        return app
+
+    def build_async_server_app(self) -> FastAPI:
+        app = FastAPI(root_path=os.getenv("TFY_SERVICE_ROOT_PATH"), docs_url="/")
+
+        app.add_api_route(
+            FUNCTION_SCHEMA_ENDPOINT,
+            lambda: get_functions_dict_with_input_signatures(self.functions),
+            methods=["GET"],
+        )
+
+        input_publisher = self.worker_config.input_config.to_input()
+        output_subscriber = self.worker_config.output_config.to_output()
+
+        if hasattr(output_subscriber, "get_output_message"):
+
+            async def get_output(request_id: str):
+                try:
+                    return await output_subscriber.get_output_message(request_id)
+                except OutputMessageTimeoutError as ex:
+                    raise HTTPException(status_code=404, detail=str(ex))
+                except MessageProcessFailure as ex:
+                    raise HTTPException(
```

Comment:
In this case I do not think this API should raise 500.

Reply:
As long as a result exists, we should return 200 and the `OutputMessage` as the body. It will have the status.